### PR TITLE
Make elastic pool id configurable on Azure SQL DB

### PR DESCRIPTION
**Pull request by mkretz**
**What this PR does / why we need it**:
We have an Azure SQL server configured with an elastic pool (not managed via Azure Service Operator). We need to create Azure SQL databases with Azure Service Operator (v1) and place them in the elastic pool. To enable this use case, we would like to make the elastic pool id configurable in `azure.microsoft.com/v1beta1/AzureSqlDatabase` by introducing a new optional attribute `elasticPoolId` in the corresponding CRD and mapping to the respective attribute in the relevant ARM data type.

**How does this PR make you feel**:
gif

**If applicable**:
- [*] this PR contains documentation
- [ ] this PR contains tests

### CODE REVIEW

**Review by matthchr — CHANGES_REQUESTED** [review]
This looks good to me. I had a few minor comments.

One ask: Can you back out the Helm chart generation? The chart you have in the PR isn't going to work as you expect, because it doesn't have an updated `image.repository` parameter in the `values.yaml`. We have an ADO job that will publish a new container image which can then be included in the Helm chart, but:
1. It has to be a 2 step process, where first the PR with the code change is merged.
2. Then, we run the ADO job, publish the image, and update the Helm chart.

So rather than trying to do them together, if you just keep your code changes in this PR, we can run the job and publish the updated container image and then update the Helm chart.

File: config/samples/azure_v1beta1_azuresqldatabase.yaml [hunk 1]
```diff
@@ -31,6 +31,9 @@ spec:
...
   # The short term retention policy to use
   # shortTermRetentionPolicy:
-      # RetentionDays is the backup retention period in days. This is how many days
-      #	Point-in-Time Restore will be supported.
+  # RetentionDays is the backup retention period in days. This is how many days
+  #	Point-in-Time Restore will be supported.
   #   retentionDays: 21
+
+  # The id of the elastic pool to place this database into. Needs to be matched with an appropriate sku
+  # elasticPoolId: /subscriptions/aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa/resourceGroups/some-resource-group-name/providers/Microsoft.Sql/servers/some-sql-server-name/elasticPools/some-elastic-pool-name
```
[the remark this answers is not in the document]
Comment (mkretz) [reply config/samples/azure_v1beta1_azuresqldatabase.yaml]: I agree. Thanks! Done.

File: pkg/resourcemanager/azuresql/azuresqlshared/sqlproperties.go [hunk 2]
```diff
@@ -37,6 +37,9 @@ type SQLDatabaseProperties struct {
 	// MaxSize - The max size of the database
 	MaxSize *resource.Quantity
 
+	// // ElasticPoolID - The resource identifier of the elastic pool containing this database.
```
[the remark this answers is not in the document]
Comment (mkretz) [reply pkg/resourcemanager/azuresql/azuresqlshared/sqlproperties.go]: Thanks! Done!

File: pkg/resourcemanager/azuresql/azuresqlshared/sqlproperties.go [hunk 3]
```diff
@@ -91,7 +94,8 @@ func SQLDatabasePropertiesToDatabase(properties SQLDatabaseProperties) (result s
 		maxSizeBytes = to.Int64Ptr(properties.MaxSize.Value())
 	}
 	result = sql.DatabaseProperties{
-		MaxSizeBytes: maxSizeBytes,
+		MaxSizeBytes:  maxSizeBytes,
+		ElasticPoolID: &properties.ElasticPoolID,
```
[the remark this answers is not in the document]
Comment (mkretz) [reply pkg/resourcemanager/azuresql/azuresqlshared/sqlproperties.go]: Thanks! I agree. I implemented a slight variant of your suggestion without the explicit assignment of `nil`. Do you agree?

[the remark this answers is not in the document]
Comment (mkretz) [reply pkg/resourcemanager/azuresql/azuresqlshared/sqlproperties.go]: I have backed out the Helm chart generation, as requested. I was unsure how this is handled. Thanks for clarifying!

[the remark this answers is not in the document]
Comment (mkretz) [reply pkg/resourcemanager/azuresql/azuresqlshared/sqlproperties.go]: @matthchr thanks for the review and comments. I have made all requested changes. Can you take another look?

**Comment by mkretz** [comment]
@matthchr thanks for the review and the comments! I have made all the requested changes. Can you take another look?

**Comment by mkretz** [comment]
@matthchr I am not sure about the failing Azure pipeline. Doesn't look like something I can fix. Correct, or am I missing something?

**Comment by matthchr** [comment]
> I am not sure about the failing Azure pipeline. Doesn't look like something I can fix. Correct, or am I missing something?

You are correct. Let me investigate what is causing this failure. It doesn't look like it's in any way related to what work you've done.

**Review by matthchr — APPROVED** [review]

**Comment by matthchr** [comment]
The issue was ours, I fixed it in #2204. Can you rebase/merge with the main branch? Once you do I can kick off another CI pass and assuming that looks good we can merge.

**Comment by mkretz** [comment]
> The issue was ours, I fixed it in #2204. Can you rebase/merge with the main branch? Once you do I can kick off another CI pass and assuming that looks good we can merge.

Thanks for fixing that! Done (rebased our fork's branch on your main).

**Comment by matthchr** [comment]
We will work on releasing an updated Helm chart for ASOv1 soon (tomorrow or early next week)

**Comment by matthchr** [comment]
https://github.com/Azure/azure-service-operator/tree/main/charts should have 1.10 now which contains this change. Try it out and let us know if there are any issues.

Thanks!

**Comment by mkretz** [comment]
> https://github.com/Azure/azure-service-operator/tree/main/charts should have 1.10 now which contains this change. Try it out and let us know if there are any issues.
> 
> Thanks!

Will do! Many thanks!

**Pull request merged**